src/main/kotlin/app/revanced/patcher/extensions/ImmutableExtensions.kt

```diff
@@ -0,0 +1,30 @@
+package app.revanced.patcher.extensions
+
+import org.jf.dexlib2.iface.instruction.ReferenceInstruction
+import org.jf.dexlib2.iface.reference.MethodReference
+import org.jf.dexlib2.immutable.reference.ImmutableMethodReference
+import java.util.*
+
+object ImmutableExtensions {
+    enum class MethodReferenceMatch {
+        DEFINING_CLASS, NAME, PARAMTER_TYPES, RETURN_TYPE;
+
+        companion object {
+            val all: EnumSet<MethodReferenceMatch> = EnumSet.allOf(MethodReferenceMatch::class.java)
+        }
+    }
+
+    fun ImmutableMethodReference.matches(
+        instruction: ReferenceInstruction,
+        match: EnumSet<MethodReferenceMatch> = MethodReferenceMatch.all
+    ): Boolean {
```

Comment:
How about inverting this and doing the following:

`myReferenceInstruction.referenEquals(otherReference)`